OpenRA.Mods.Common/Traits/World/Pathfinder/PathSearch.cs

```diff
@@ -0,0 +1,197 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+using OpenRA.Primitives;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public sealed class PathSearch : BasePathSearch
+	{
+		public override CellLayer<CellInfo> CellInfo { get; protected set; }
+
+		public override IEnumerable<Pair<CPos, int>> Considered
+		{
+			get { return considered; }
+		}
+
+		LinkedList<Pair<CPos, int>> considered;
```

Comment:
Why is it so? List requires resizing the internal array everytime you add a new element. Adding elements in a linkedlist is instantly without having to resize. LinkedList are like Lists which do not allow random access (not necessary for this case), but offer faster deletions and insertions in exchange. 

The right structure for the right job IMHO.